src/main/java/org/embulk/input/zendesk/ZendeskInputPlugin.java

```diff
@@ -280,64 +298,143 @@ private JsonNode buildColumns(final PluginTask task)
                     }
-                    entry.put("type", Types.LONG.getName());
+                    entry.put("type", Types.STRING.getName());
                 }
+            }
+            addIncludedObjectsToSchema((ArrayNode) columns, includes);
+            return columns;
+        }
+        else {
+            throw new ConfigException("Fail to guess column");
+        }
+    }
 
-                // Id of User Events target is more suitable for String
-                if (target.equals(Target.USER_EVENTS)) {
-                    entry.put("type", Types.STRING.getName());
+    @VisibleForTesting
+    protected ConfigDiff guessData(final JsonNode jsonNode, final String targetJsonName)
+    {
+        List<String> unifiedFieldNames = unifiedFieldNames(jsonNode, targetJsonName);
+        List<List<Object>> samples = createSampleData(jsonNode, targetJsonName, unifiedFieldNames);
+        List<ConfigDiff> columnConfigs = SchemaGuess.of(CONFIG_MAPPER_FACTORY).fromListRecords(unifiedFieldNames, samples);
+        if (columnConfigs.isEmpty()) {
+            throw new ConfigException("Fail to guess column");
+        }
+        columnConfigs.forEach(conf -> conf.remove("index"));
+        ConfigDiff parserGuessed = CONFIG_MAPPER_FACTORY.newConfigDiff();
+        parserGuessed.set("columns", columnConfigs);
+
+        ConfigDiff configDiff = CONFIG_MAPPER_FACTORY.newConfigDiff();
+        configDiff.setNested("parser", parserGuessed);
+        return configDiff;
+    }
+
+    private List<List<Object>> createSampleData(JsonNode jsonNode, String targetJsonName, List<String> unifiedFieldNames)
+    {
+        final List<List<Object>> samples = new ArrayList<>();
+        Iterator<JsonNode> records = ZendeskUtils.getListRecords(jsonNode, targetJsonName);
+        while (records.hasNext()) {
+            JsonNode node = records.next();
+            List<Object> line = new ArrayList<>();
+            for (String field : unifiedFieldNames) {
+                JsonNode childNode = node.get(field);
+                if (childNode == null || childNode.isNull() || "null".equals(childNode.asText())) {
+                    line.add(null);
+                    continue;
+                }
+                if (childNode.isContainerNode()) {
+                    line.add(childNode);
+                }
+                else {
+                    line.add(childNode.asText());
                 }
             }
-            else if (idPattern.matcher(name).find()) {
-                if (type.equals(Types.TIMESTAMP.getName())) {
-                    entry.remove("format");
+            samples.add(line);
+        }
+        return samples;
+    }
+
+    private List<String> unifiedFieldNames(JsonNode jsonNode, String targetJsonName)
+    {
+        List<String> columnNames = new ArrayList<>();
+        Iterator<JsonNode> records = ZendeskUtils.getListRecords(jsonNode, targetJsonName);
+        while (records.hasNext()) {
```

Comment:
I think for unified field name, we don't need to go through all the records, just one or a few ones are enough

Reply:
As our discussion, I test it on development env and it work well with page (~100 records)